test/jobs/maintenance_tasks/task_job_test.rb

```diff
@@ -64,6 +64,22 @@ class TaskJobTest < ActiveJob::TestCase
       assert_no_enqueued_jobs
     end
 
+    test ".perform_now skips iterations when Run is cancelled even when a race condition happens" do
```

Comment:
Test name is a little unclear, maybe:
```suggestion
    test ".perform_now avoids iterating and cancels Run when race occurs between starting and cancelling a Run" do
```
?

Reply:
Right, I used that with symmetry with the previous test. Updated both.